utils/include/XMLParser.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+#include "stdafx.h"
+
+class XMLParser
+{
+public:
+	XMLParser() = delete;
+	XMLParser(XMLParser&) = delete;
+	XMLParser(XMLParser&&) = delete;
+
+	XMLParser(std::string path_to_configuration_file) :
+		path_to_configuration_file_(path_to_configuration_file)
+	{ };
+
+	~XMLParser() = default;
+
+	int getIntegerConfiguration(const std::string& data_path) const;
+	bool isConfigurationEnabled(const std::string& data_path) const;
+	std::string getStringConfiguration(const std::string& data_path) const;
```

Comment:
std::string getStringConfiguration(const std::string& data_path) const; -> bool/void getStringConfiguration(const std::string& data_path, string & returnStr); 
The same to other methods in this class. When you return a string there is a moment of time when 2 copies of the string are present. Imagine if the string has 100MB it doubles for  a few nano sec and we might  be out of memory.